pysaliency/filter_datasets.py

```diff
@@ -232,3 +232,50 @@ def filter_stimuli_by_size(stimuli, fixations, size=None, sizes=None):
     indices = [i for i in range(len(stimuli)) if stimuli.sizes[i] in sizes]
 
     return create_subset(stimuli, fixations, indices)
+
+
+def filter_scanpaths_by_attribute(scanpaths:  FixationTrains, attribute_name, attribute_value, invert_match=False):
+    """Filter Scanpaths by values of scanpath attribute (fixation_trains.scanpath_attributes)"""
+    
+    mask = np.array([element == attribute_value for element in getattr(scanpaths, attribute_name)])
+    if invert_match is True:
+        mask = ~mask
+    indices = list(np.nonzero(mask)[0])
```

Comment:
Do you need indices here? I thought the mask directly should work in filter_fixation_trains? But maybe I'm mistaken. It should definitely work for `Fixations` though

Reply:
yes it works for `Fixaions` but for `FixationTrains` we need the indices

Reply:
are you sure? I just tested `scanpaths.filter_fixation_trains(scanpaths.train_ns==0)` and it worked well.